internal/filter/timeFrame.go

```diff
@@ -0,0 +1,22 @@
+package filter
+
+import (
+	"github.com/inovex/CalendarSync/internal/models"
+)
+
+type TimeFrameEvents struct {
+	HourStart int
+	HourEnd   int
+}
+
+func (a TimeFrameEvents) Name() string {
+	return "TimeFrame"
+}
+
+func (a TimeFrameEvents) Filter(event models.Event) bool {
+	// If the user enters invalid hours, such as numbers higher than 24 or lower than 0, all events will be synchronized up to that point in time. For example, if the start hour is -1 and the end hour is 17, all events until 17 o'clock will be synchronized.
+	if event.StartTime.Hour() >= a.HourStart && event.StartTime.Hour() <= a.HourEnd {
```

Comment:
I was wondering about the exact semantics we want here. Assuming a time frame from 8-17 we currently get the following behavior:

event start/end | filter
--- | ---
6-7 | false
7-9 | false (why?)
10-12 | true
16-18 | true
18-19 | false
6-18 | false (why?)


I think we want to include all events that overlap with the timeFrame in any way. That is, an event should be included if the event start or end is within the timeFrame but also the other way around if the timeFrame start or end are within the event.

Reply:
> I think we want to include all events that overlap with the timeFrame in any way.

yup, that should be the expected behaviour. Implemented the logic like this and added some more test cases.